packages/pluggableWidgets/datagrid-web/src/components/Header.tsx

```diff
@@ -44,24 +44,21 @@ export function Header<D extends object>(props: HeaderProps<D>): ReactElement {
         : undefined;
 
     return (
-        <div
-            className="th"
+        <td
+            className={classNames("th", canDrag && props.column.id === props.dragOver ? "dragging" : "")}
             {...rest}
             style={{
                 ...style,
-                ...(!props.resizable ? { flex: "1 1 0px" } : {}),
                 ...(!props.sortable || !props.column.canSort ? { cursor: "unset" } : {})
             }}
             title={props.column.render("Header") as string}
+            {...(props.column.weight !== 1
```

Comment:
Can't you just have `width={props.column.weight > 1 ? `${props.column.weight}%` : props.column.weight === 0 ? "1px" : undefined`?

Reply:
Done